### ISSUE TITLE
cookie.delete() should accept an options argument so you can pass `path`

### ISSUE THREAD

**Issue by danawoodman** [sveltejs/kit#6817]
### Describe the problem

(_Sorry if this is a know issue/topic, I searched by couldn't find anything._)

Cannot remove a cookie using `cookies.delete` on a path (say `src/routes/foo/+page.server.ts`) if the cookie was made at another path (say `src/routes/+layout.server.ts`) since `path` gets set to `/` and calling `delete` in `/foo` will pass `path: "/foo"`.

### Describe the proposed solution

Add an optional argument paramter to `.delete` to accept options the same as `set()`:

```ts
cookies.delete("auth_token", { path: "/" })
```

### Alternatives considered

Using set to empty the cookie but that is a hack:

```ts
cooklies.set("auth_token", "", { path: / })
```

### Importance

would make my life easier

### Additional Information

```ts
// src/routes/logout/+page.server.ts
import { redirect } from "@sveltejs/kit";
import type { PageServerLoad } from "./$types";

export const load: PageServerLoad = async (event) => {
	event.cookies.delete("auth_token");
	throw redirect(302, "/");
};
```

```ts
// src/routes/login/+page.server.ts
import { invalid, redirect } from "@sveltejs/kit";
import type { Actions } from "./$types";

export const actions: Actions = {
	async default(event) {
		// Some auth logic here...

		event.cookies.set("auth_token", token, { path: "/" });

		throw redirect(302, "/dashboard");
	},
};
```

**Comment by alialaa** [comment]
I had this issue but I believe it is fixed in [483](https://github.com/sveltejs/kit/releases/tag/%40sveltejs%2Fkit%401.0.0-next.483). I haven't tried yet though.

**Comment by s3812497** [comment]
Hey Dana, I've just learnt that this has recently been added. I've filed a similar issue here #6609 and was recommended this exact solution.
https://github.com/sveltejs/kit/issues/6609#issuecomment-1247425220

**Comment by dummdidumm** [comment]
Fixed via #6811

**Comment by danawoodman** [comment]
Nice! Thanks!
